packages/sdk/src/utils/executeSteps.ts

```diff
@@ -73,6 +78,19 @@ export async function executeSteps(
     // Handle errors
     if (json.error || !json.steps) throw json
 
+    // Check if step's transactions can be batched and if wallet supports atomic batch
+    // If so, manipulate steps to batch transactions
+    if (canBatchTransactions(json.steps)) {
+      isAtomicBatchSupported = Boolean(
+        wallet?.supportsAtomicBatch &&
+          (await wallet?.supportsAtomicBatch(chainId))
+      )
+      if (isAtomicBatchSupported) {
+        const batchedStep = prepareBatchTransaction(json.steps)
+        json.steps = [batchedStep, ...json.steps.slice(2)]
```

Comment:
I don't love this, but I understand why we need it. Why are we keeping some of the steps though?

Reply:
Yeah I suppose we can update this line to:
`json.steps = [batchedStep]`

Thoughts?

Reply:
yeah makes sense to me.